sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2002,6 +2002,20 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             Context->Devices[0]->ZeDeviceProperties.flags &
                                 ZE_DEVICE_PROPERTY_FLAG_INTEGRATED;
 
+  // PI_MEM_FLAGS_HOST_PTR_ALLOC flag indicates allocation of pinned
+  // host memory which is accessible from device.
+  bool AllocHostPtr = Flags & PI_MEM_FLAGS_HOST_PTR_ALLOC;
+
+  if (AllocHostPtr) {
+    assert(HostPtr == nullptr &&
+           "PI_MEM_FLAGS_HOST_PTR_ALLOC cannot be used with host pointer");
+
+    ze_host_mem_alloc_desc_t ZeDesc = {};
+    ZeDesc.flags = 0;
+
+    ZE_CALL(zeMemAllocHost(Context->ZeContext, &ZeDesc, Size, 1, &Ptr));
+  }
```

Comment:
This can't be right. The code below will do another memory allocation for `Ptr` effectively leaking memory allocated in line 2016. Given the fact that there is no test failed, I am asking again that a test be added, which checks the desired behavior intended to be added here.

Reply:
@rbegam ping

Reply:
@smaslov-intel I was under the impression that these two cases are mutually exclusive. I'll follow up on our discussion and update it.